DataFormats/TrackingRecHitSoA/interface/TrackingRecHitsDevice.h

```diff
@@ -0,0 +1,43 @@
+#ifndef DataFormats_TrackingRecHitSoA_interface_TrackingRecHitSoADevice_h
+#define DataFormats_TrackingRecHitSoA_interface_TrackingRecHitSoADevice_h
+
+#include <cstdint>
+
+#include <alpaka/alpaka.hpp>
+
+#include "DataFormats/Portable/interface/PortableDeviceCollection.h"
+#include "DataFormats/TrackingRecHitSoA/interface/TrackingRecHitsHost.h"
+#include "DataFormats/TrackingRecHitSoA/interface/TrackingRecHitsSoA.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+template <typename TrackerTraits, typename TDev>
+class TrackingRecHitDevice : public PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>, TDev> {
+public:
+  using hitSoA = TrackingRecHitSoA<TrackerTraits>;
+  //Need to decorate the class with the inherited portable accessors being now a template
+  using PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>, TDev>::view;
+  using PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>, TDev>::const_view;
+  using PortableDeviceCollection<TrackingRecHitLayout<TrackerTraits>, TDev>::buffer;
+
+  TrackingRecHitDevice() = default;
+
+  // Constructor which specifies the SoA size
+  template <typename TQueue>
+  explicit TrackingRecHitDevice(uint32_t nHits, int32_t offsetBPIX2, uint32_t const* hitsModuleStart, TQueue queue)
```

Comment:
```suggestion
  explicit TrackingRecHitDevice(uint32_t nHits, int32_t offsetBPIX2, uint32_t const* hitsModuleStart, TQueue& queue)
```
or perhaps move the `queue` as first sine that is the Alpaka's convention (that we'd likely want to move to elsewhere too)
```suggestion
  explicit TrackingRecHitDevice(TQueue& queue, uint32_t nHits, int32_t offsetBPIX2, uint32_t const* hitsModuleStart)
```

Or did you have a reason to copy the Queue? (this same comment applies to the earlier data formats as well, I won't comment there).